### PR TITLE
ElasticSearch Update

### LINKED ISSUE #64: ElasticSearchStorage does not work with Elasticsearch 6
The template itself doesn't work and throws some errors. But even with all the cuckoo stuff removed it still throws this error.

`ERROR: storage ElasticSearchStorage failed to load. TransportError(400, 'illegal_argument_exception', 'Rejecting mapping update to [multiscanner_reports] as the final mapping would have more than 1 type: [note, report]')`

### LINKED ISSUE #74: Make ES storage module more resilient
If the ES server is down, multiscanner will error out and for example, the API will be unusable. It should retry the connection periodically to be more resilient. This will help in docker (will remove the need for a "sleep" command to wait for the ES container to be up and running).

### LINKED ISSUE #117: Results page should return error if task_id does not exist
Currently, report page will respond with 'Scan result pending...' indefinitely if the task_id does not exist. Page should return error immediately for non-existent task_ids.

### LINKED ISSUE #120: AttributeError caused by exception in ElasticSearchStorage.get_report
### Description
If `ElasticSearchStorage.get_report` raises an exception, the method returns `None` (see [here](https://github.com/mitre/multiscanner/blob/master/storage/elasticsearch_storage.py#L304)). This causes an `AttributeError` in the REST API.

#### Error

```
api_1      |   File "utils/api.py", line 576, in get_report
api_1      |     report_dict = _pre_process(report_dict)
api_1      |   File "utils/api.py", line 589, in _pre_process
api_1      |     if report_dict.get('Report', {}).get('ssdeep', {}):
api_1      | AttributeError: 'NoneType' object has no attribute 'get'
```

### PR DESCRIPTION
**Pull request by clenk**
- Upgrade to ES version 6
- Clean up ES template
- Make API/storage more resilient (retry to connect when starting up)
- Remove "doc_type" as a configuration option
- Make Report page display an error if task_id does not exist in task database
- Fix bug when getting non-existent report from ElasticSearch

Fixes #64, fixes #74,  fixes #117, fixes #120.

### CODE REVIEW

**Comment by awest1339** [comment]
We should use the ElasticSearch recommended doc type of "_doc". From here: https://www.elastic.co/guide/en/elasticsearch/reference/6.x/removal-of-types.html

File: storage/storage.py [hunk 1]
```diff
@@ -126,19 +137,70 @@ def __init__(self, configfile=CONFIG, config=None, configregen=False):
...
-        if loaded_storage == []:
-            raise RuntimeError('No storage classes loaded')
-        self.loaded_storage = loaded_storage
+        self.storage_classes = storage_classes
+        self.loaded_storage = {}
+
+        # Setup each enabled storage
+        self.load_modules()
+
+    def load_modules(self, required_module=""):
```
[the remark this answers is not in the document]
Comment (ptcNOP) [reply storage/storage.py]: I guess it only returns a module if `required_module` is set.

File: storage/storage.py [hunk 2]
```diff
@@ -126,19 +137,70 @@ def __init__(self, configfile=CONFIG, config=None, configregen=False):
...
+                storage_error = None
+
+            if storage_error:
+                time.sleep(self.sleep_time)
+            else:
+                break
+
+        if storage_error:
+            if required_module:
+                sys.exit('ERROR: {} module not loaded!'.format(required_module))
```
Comment (ptcNOP) [inline storage/storage.py]: Maybe raise our own exception instead of `SystemExit`...?

**Comment by ptcNOP** [comment]
I had a couple of comments about the new method, `load_modules`, but other than that, these changes are working for me.

**Comment by ptcNOP** [comment]
I like these changes. This looks good to me.

**Pull request merged**